packages/core/src/components/event-lake.aggregate-store.suite.ts

```diff
@@ -0,0 +1,336 @@
+    .auto(Withdrawn)
+    .auto(Account)
+    .auto(Bank);
+
+  const bankStore = config.getStore("Bank", AutoSerializer.first(Bank));
+
+  it("save and load", async () => {
+    const accountStore = config.getAggregateStore(Account, registry);
+    const account = Account.open();
+    await accountStore.save(account);
+    const loadedAccount = await accountStore.load(account.id);
+    expect(loadedAccount).toEqual(account);
+    expect(loadedAccount?.balance).toEqual(0);
+  });
+
+  it("supports concurrent writes, using transactions", async () => {
+    const accountStore = config.getAggregateStore(Account, registry);
+    const account = Account.open();
+    await accountStore.save(account);
+
+    await Promise.all(
+      [...Array(5).keys()].map(async () => {
+        await config.transaction.perform(async (trx) => {
+          const fresh = await accountStore.load(account.id, trx);
+          fresh!.deposit(1);
+          await accountStore.save(fresh!, trx);
+        });
+      }),
+    );
+
+    const loadedAccount = await accountStore.load(account.id);
+    expect(loadedAccount?.balance).toEqual(5);
+  });
+
+  it("is fast", async () => {
+    const accountStore = config.getAggregateStore(Account, registry);
+
+    const accounts = [...Array(200).keys()].map(() => Account.open());
+
+    const expectMS = async (ms: number, fn: () => Promise<any>) => {
+      const before = process.hrtime.bigint();
+      await fn();
+      const after = process.hrtime.bigint();
+      expect(Number(after - before) / 1_000_000).toBeLessThan(ms);
+    };
+
+    await expectMS(5000, async () => {
+      for (const account of accounts) {
+        await accountStore.save(account);
+      }
+    });
+
+    await expectMS(5000, async () => {
+      for (const account of accounts) {
+        await accountStore.load(account.id);
+      }
+    });
+  });
+
+  it("supports saveAll with transactions", async () => {
+    const accountStore = config.getAggregateStore(Account, registry);
+
+    const accounts = [...Array(10).keys()].map(() => Account.open());
+    await accountStore.saveAll(accounts);
+
+    const loadedAccounts = await Promise.all(
+      accounts.map((account) => accountStore.load(account.id)),
+    );
+
+    expect(loadedAccounts).toEqual(accounts);
+  });
+
+  class MockEventBus implements IEventBus {
+    off(): void {
+      throw new Error("Method not implemented.");
+    }
+    on() {
+      throw new Error("Method not implemented.");
+    }
+
+    events: IEvent[] = [];
+
+    publish(event: IEvent) {
+      this.events.push(event);
+      return Promise.resolve();
+    }
+  }
+  it("should support saveAll with transactions", async () => {
+    const eventBus = new MockEventBus();
+    const accountStore = config.getAggregateStore(Account, registry, eventBus);
+
+    const bank = Bank.new();
+    await bankStore.save(bank);
+
+    let willFailForAttempt = 3;
+    const saved = await config.transaction.perform(async (trx) => {
```

Comment:
Just for my knowledge, transactions are retried by default ? :)

Reply:
transactions are retried 5 time by default, but you can override it